app/components/concept.ts

```diff
@@ -157,7 +180,7 @@ export default class ConceptComponent extends Component<Signature> {
 
   updateLastRevealedBlockGroupIndex(newBlockGroupIndex: number) {
     this.lastRevealedBlockGroupIndex = newBlockGroupIndex;
-    this.args.onProgressPercentageChange(this.progressPercentage);
+    this.updateProgressPercentage(this.progressPercentage);
```

Comment:
It's a bit weird that we're passing in `this.progressPercentage` now, since the `updateProgressPercentage` method already has access to `this.progressPercentage`. 

It'a also confusing that there's `conceptEngagement.progressPercentage` and `progressPercentage`, and it's not super clear how they're different. 

How about this: 

- Remove `this.progressPercentage` from this method call, the method already has access to `this.progressPercentage`
- Name the method `updateConceptEngagement` instead of `updateProgressPercentage`
- Rename `this.progressPercentage` to `this.computedProgressPercentage`, that way it's clear that we call this when we want to "compute", and it's actually stored on `conceptEngagement.progressPercentage`.